datameta/storage.py

```diff
@@ -35,7 +35,9 @@ def demo_mode(request):
 
 def rm(request, storage_path):
     """Remove a file from storage by local storage file name"""
-    if not demo_mode(request):
+    if not storage_path:
+        log.debug("NO STORAGE PATH. THIS SHOULD ONLY HAPPEN IN TEST MODE.")        
+    elif not demo_mode(request):
```

Comment:
Let's not touch the main code to make tests work. We have files available in the fixtures, we also have a storage path that is automatically cleaned up after tests are run, so let's make use of that and copy the file to the storage path (see below).
```suggestion
    if not demo_mode(request):
```

Reply:
tbh I thought that this caught a case that is not handled by the `rm` function, i.e. if `storage_path` is `None` for whatever reason. (in this case this reason was my test, but I don't know if this could occur somehow.)

Reply:
Thanks for the upload-part, @lkuchenb!